src/gui/advancedsettings.cpp

```diff
@@ -200,7 +200,7 @@ void AdvancedSettings::saveAdvancedSettings()
     }
     session->setOSMemoryPriority(prio);
 
-    static_cast<Application *>(QCoreApplication::instance())->setMemoryWorkingSetLimit(m_spinBoxMemoryWorkingSetLimit.value());
+    dynamic_cast<IApplication *>(QCoreApplication::instance())->setMemoryWorkingSetLimit(m_spinBoxMemoryWorkingSetLimit.value());
```

Comment:
Please don't use `dynamic_cast` unless you really need to test its result for `nullptr`. But it isn't needed since you obviously know that it implements IApplication (by design).

Reply:
How can I do it without `dynamic_cast`?
`QCoreApplication::instance()` first needs to downcast to `Application *` then upcast to `IApplication *`.
`static_cast<IApplication *>(QCoreApplication::instance())` won't work as compiler complains.
`reinterpret_cast<IApplication *>(QCoreApplication::instance())` will result in crash.

Reply:
Well. Let's leave `dynamic_cast` for now. Anyway, I have the idea of a further redesign that will make the relationship of the components more explicit.